### ISSUE TITLE
Support `IDENTITY` instead of serial columns

### ISSUE THREAD

**Issue by zachdaniel** [ash-project/ash_postgres#233]
Instead of the bigserial type, we should use `GENERATED BY DEFAULT AS IDENTITY`. For more information, see here: https://wiki.postgresql.org/wiki/Don%27t_Do_This#Don.27t_use_serial

We can put this behind an opt-in configuration until the next major release.

**Comment by Munksgaard** [comment]
I'm interested in working on this. Would you prefer this to be a separate type, such as the `:identity` type that is supported by ecto, or should it be a configuration that makes `generated? true` use this instead of a `bigserial`?

**Comment by Munksgaard** [comment]
And how should it interact with [`integer_primary_key`](https://hexdocs.pm/ash/dsl-ash-resource.html#attributes-integer_primary_key)?

**Comment by zachdaniel** [comment]
Yeah, I think it would be something like `identity_column_defaults [:id]` since otherwise the type is an integer for all other purposes.

**Comment by yenrab** [comment]
Did this get merged and not closed?
